src/main/java/com/exasol/adapter/dynamodb/mapping/SchemaMappingDefinitionToSchemaMetadataConverter.java

```diff
@@ -51,17 +61,75 @@ private ColumnMetadata convertColumn(final AbstractColumnMappingDefinition colum
                 .adapterNotes(serialized).build();
     }
 
+    /**
+     * Deserializes a {@link TableMappingDefinition} from {@link TableMetadata}.
+     *
+     * @param tableMetadata  metadata for the table to be deserialized
+     * @param schemaMetadata needed because the tables can't be serialized into the TableMetadata due to a bug
+     * @return deserialized {@link TableMappingDefinition}
+     * @throws IllegalStateException if deserialization fails
+     */
+    public TableMappingDefinition convertBackTable(final TableMetadata tableMetadata,
+            final SchemaMetadata schemaMetadata) {
+        try {
+            return convertBackTableIntern(tableMetadata, schemaMetadata);
+        } catch (final IOException | ClassNotFoundException exception) {
+            throw new IllegalStateException("Failed to deserialize TableMappingDefinition.", exception);
+        }
+    }
+
+    private TableMappingDefinition convertBackTableIntern(final TableMetadata tableMetadata,
+            final SchemaMetadata schemaMetadata) throws IOException, ClassNotFoundException {
+        final TableMappingDefinition preliminaryTable;
+        preliminaryTable = findTableInSchemaMetadata(tableMetadata.getName(), schemaMetadata);
```

Comment:
Inline

Reply:
the comment? if I use // comments the formater splits it to ten lines or so...